toolkit/tools/pkg/imagecustomizerlib/imagecustomizer_test.go

```diff
@@ -195,6 +208,193 @@ func TestValidateConfigScriptNonExecutable(t *testing.T) {
+	}
+
+	buildDir := filepath.Join(tmpDir, "TestCustomizeImageKernelCommandLine")
+	outImageFilePath := filepath.Join(buildDir, "image.vhd")
+
+	// Create fake disk.
+	diskFilePath, err := createFakeEfiImage(buildDir)
+	if !assert.NoError(t, err) {
+		return
+	}
+
+	// Customize image.
+	config := &imagecustomizerapi.Config{
+		Disks: &[]imagecustomizerapi.Disk{
+			{
+				PartitionTableType: imagecustomizerapi.PartitionTableTypeGpt,
+				MaxSize:            4096,
+				Partitions: []imagecustomizerapi.Partition{
+					{
+						ID:     "boot",
+						Flags:  []imagecustomizerapi.PartitionFlag{"esp", "boot"},
+						Start:  1,
+						End:    ptrutils.PtrTo(uint64(9)),
+						FsType: "fat32",
+					},
+					{
+						ID:     "rootfs",
+						Start:  9,
+						End:    ptrutils.PtrTo(uint64(2048)),
+						FsType: "ext4",
+					},
+					{
+						ID:     "var",
+						Start:  2048,
+						FsType: "xfs",
+					},
+				},
+			},
+		},
+		SystemConfig: imagecustomizerapi.SystemConfig{
+			BootType: imagecustomizerapi.BootTypeEfi,
+			PartitionSettings: []imagecustomizerapi.PartitionSetting{
+				{
+					ID:           "boot",
+					MountPoint:   "/boot/efi",
+					MountOptions: "umask=0077",
+				},
+				{
+					ID:         "rootfs",
+					MountPoint: "/",
+				},
+				{
+					ID:         "rootfs",
+					MountPoint: "/var",
+				},
+			},
+			KernelCommandLine: imagecustomizerapi.KernelCommandLine{
+				ExtraCommandLine: "console=tty0 console=ttyS0",
+			},
+		},
+	}
+
+	err = CustomizeImage(buildDir, buildDir, config, diskFilePath, nil, outImageFilePath, "raw", false)
+	if !assert.NoError(t, err) {
+		return
+	}
+
+	// Mount the output disk image so that its contents can be checked.
+	imageConnection := NewImageConnection()
+	defer imageConnection.Close()
+
+	err = imageConnection.ConnectLoopback(outImageFilePath)
+	if !assert.NoError(t, err) {
+		return
+	}
+
+	rootDir := filepath.Join(buildDir, testImageRootDirName)
+
+	bootPartitionDevPath := fmt.Sprintf("%sp1", imageConnection.Loopback().DevicePath())
+	osPartitionDevPath := fmt.Sprintf("%sp2", imageConnection.Loopback().DevicePath())
+	varPartitionDevPath := fmt.Sprintf("%sp3", imageConnection.Loopback().DevicePath())
+
+	mountPoints := []*safechroot.MountPoint{
+		safechroot.NewPreDefaultsMountPoint(osPartitionDevPath, "/", "ext4", 0, ""),
+		safechroot.NewMountPoint(bootPartitionDevPath, "/boot/efi", "vfat", 0, ""),
+		safechroot.NewMountPoint(varPartitionDevPath, "/var", "xfs", 0, ""),
+	}
+
+	err = imageConnection.ConnectChroot(rootDir, false, []string{}, mountPoints)
+	if !assert.NoError(t, err) {
+		return
+	}
+
+	// Read the grub.cfg file.
+	grub2ConfigFilePath := filepath.Join(imageConnection.Chroot().RootDir(), "/boot/grub2/grub.cfg")
+
+	grub2ConfigFile, err := os.ReadFile(grub2ConfigFilePath)
+	if !assert.NoError(t, err) {
+		return
+	}
+
+	t.Logf("%s", grub2ConfigFile)
+
+	linuxCommandLineRegex, err := regexp.Compile(`linux .* console=tty0 console=ttyS0 `)
```

Comment:
Thoughts on expanding this so we can double check it doesn't mangle the rest of the command line we want to keep? Bit hacky but if the image config uses PARTLABEL I think the whole line should be deterministic.

Reply:
I added the two surrounding values.